alerter/src/main/java/com/tapadoo/alerter/Alert.kt

```diff
@@ -0,0 +1,576 @@
+import android.support.v4.view.ViewCompat
+import android.support.v7.content.res.AppCompatResources
+import android.text.TextUtils
+import android.util.AttributeSet
+import android.util.Log
+import android.view.HapticFeedbackConstants
+import android.view.MotionEvent
+import android.view.View
+import android.view.ViewGroup
+import android.view.animation.Animation
+import android.view.animation.AnimationUtils
+import android.view.animation.LinearInterpolator
+import android.widget.FrameLayout
+import android.widget.LinearLayout
+import kotlinx.android.synthetic.main.alerter_alert_view.view.*
+
+/**
+ * Custom Alert View
+ *
+ * @author Kevin Murphy, Tapadoo, Dublin, Ireland, Europe, Earth.
+ * @since 26/01/2016
+ */
+class Alert @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = 0)
+    : FrameLayout(context, attrs, defStyle), View.OnClickListener, Animation.AnimationListener, SwipeDismissTouchListener.DismissCallbacks {
+
+    internal var onShowListener: OnShowAlertListener? = null
+    internal var onHideListener: OnHideAlertListener? = null
+
+    internal var slideInAnimation: Animation? = null
+    internal var slideOutAnimation: Animation? = null
+
+    internal var duration = DISPLAY_TIME_IN_SECONDS
+
+    private var enableIconPulse = true
+    private var enableInfiniteDuration: Boolean = false
+    private var enableProgress: Boolean = false
+
+    private var runningAnimation: Runnable? = null
+
+    private var isDismissable = true
+
+    /**
+     * Flag to ensure we only set the margins once
+     */
+    private var marginSet: Boolean = false
+
+    /**
+     * Flag to enable / disable haptic feedback
+     */
+    private var vibrationEnabled = true
+
+    /**
+     * Sets the Gravity of the Alert
+     *
+     * @param contentGravity Gravity of the Alert
+     */
+    var contentGravity: Int
+        get() = (rlContainer!!.layoutParams as FrameLayout.LayoutParams).gravity
+        set(contentGravity) {
+            val paramsTitle = tvTitle?.layoutParams as LinearLayout.LayoutParams
+            paramsTitle.gravity = contentGravity
+            tvTitle?.layoutParams = paramsTitle
+
+            val paramsText = tvText?.layoutParams as LinearLayout.LayoutParams
+            paramsText.gravity = contentGravity
+            tvText?.layoutParams = paramsText
+        }
+
+    init {
+        initView()
+    }
+
+    private fun initView() {
+        inflate(context, R.layout.alerter_alert_view, this)
+        isHapticFeedbackEnabled = true
+
+        ViewCompat.setTranslationZ(this, Integer.MAX_VALUE.toFloat())
+
+        flAlertBackground.setOnClickListener(this)
+
+        //Setup Enter & Exit Animations
+        slideInAnimation = AnimationUtils.loadAnimation(context, R.anim.alerter_slide_in_from_top)
+        slideOutAnimation = AnimationUtils.loadAnimation(context, R.anim.alerter_slide_out_to_top)
+        slideInAnimation!!.setAnimationListener(this)
```

Comment:
If you move this to the init block, you can remove the nullable-ness of these objects.